src/types.rs

```diff
@@ -1580,25 +1923,137 @@ impl Rule {
             None
         }
     }
-    /// Match one rule against another.
+    /// Pattern match one `Rule` against another.
+    /// 
+    /// See [`Pattern Matching`] for more information.
```

Comment:
I would collapse these two sentences into one and say ``[`Pattern Match`] one `Rule` against another.`` Do you see any reason not to do that? If not, go ahead and change it.